File: playground/tools.js
import { python, shutdown } from "../index.js";

const { greet, total, Counter, multiply, factorial } = await python("./playground/tools.py");

console.log(await greet("Mayowa"));
console.log(await total([10, 20, 30]));
console.log(await multiply(4,6));
console.log(await factorial(5));
const counter = await Counter(5);
console.log(await counter.add(3));

await counter.release();
shutdown();

File: playground/tools.py
def greet(name):
    return f"Hello, {name}!"

def total(numbers):
    return sum(numbers)

def multiply(a,b):
    return a * b

def factorial(n):
    if n == 0:
        return 1
    else:
        return n * factorial(n-1)

class Counter:
    def __init__(self, start=0):
        self.value = start

    def add(self, amount=1):
        self.value += amount
        return self.value